scipy/ndimage/tests/test_interpolation.py

```diff
@@ -432,22 +533,24 @@ def mapping(x):
         # expected result.
         npad = 24
         pad_mode = ndimage_to_numpy_mode.get(mode)
-        xp = np.pad(x, npad, mode=pad_mode)
+        x_padded = xp.asarray(np.pad(np.asarray(x), npad, mode=pad_mode))
```

Comment:
no need to create with `xp` and go straight to `np`, should just start with `np` and convert once